packages/server/graphql/private/types/SetIsFreeMeetingTemplateSuccess.ts

```diff
@@ -0,0 +1,20 @@
+import getRethink from '../../../database/rethinkDriver'
+import {ResolversTypes, SetIsFreeMeetingTemplateSuccessResolvers} from '../../private/resolverTypes'
+
+export type SetIsFreeMeetingTemplateSuccessSource = {
+  updatedTemplateIds: string[]
+}
+
+const SetIsFreeMeetingTemplateSuccess: SetIsFreeMeetingTemplateSuccessResolvers = {
+  updatedTemplates: async ({updatedTemplateIds}) => {
+    if (!updatedTemplateIds) return null
+    const r = await getRethink()
+    const updatedTemplates = (await r
+      .table('MeetingTemplate')
+      .getAll(r.args(updatedTemplateIds))
+      .run()) as ResolversTypes['MeetingTemplate'][]
```

Comment:
I'm not sure why the casting is required here as `updatedTemplates` is defined as `[MeetingTemplate!]` in gql and the returned value from this rethinkdb query is `MeetingTemplate[]`

Reply:
-1 This is the TS error I get
```
 Type 'MeetingTemplate' is not assignable to type 'Omit<ReflectTemplate, "team" | "prompts">'.
        Types of property 'isFree' are incompatible.
          Type 'boolean | undefined' is not assignable to type 'boolean'.
            Type 'undefined' is not assignable to type 'boolean'. (tsserver 2322)
```
so it might be worth fixing the type of `isFree` now before we have to add more such casts.

Reply:
Hey @mattkrick, this is the TS error from yesterday's office hours. After updating the `isFree` field, I get that ugly error message starting with:

```
(property) updatedTemplates?: Resolver<Maybe<(ResolverTypeWrapper<Omit<PokerTemplate, "team" | "dimension" | "dimensions"> & {
dimension: ResolverTypeWrapper<Omit<TemplateDimension, "team" | "template" | "selectedScale"> & {
selectedScale: ResolverTypeWrapper<Omit<TemplateScale, "team" | "dimensions"> & {
```

You mentioned that I should send over a link and you might spend a few minutes digging into it? No worries if not!

Reply:
sure, i'll take a look now

Reply:
here's the fix, let me know if that makes sense!
https://www.loom.com/share/4aea62d9a3ce4c918766ec824ad10bfa

Reply:
Fantastic, thanks for the explanation on mappers and why it's better to return an empty array vs null in the success case! It's always helpful to see how you got to the solution